hydra-node/src/Hydra/Ledger/Cardano.hs

```diff
@@ -474,9 +624,41 @@ fromCardanoApiUtxo :: Cardano.Api.UTxO AlonzoEra -> Utxo
 fromCardanoApiUtxo = coerce
 
 --
--- KeyWitness
+-- Witnesses
 --
 
+-- TODO: This could be made available upstream...
```

Comment:
But we don't need it now, right? We could just always create a `Witness ScriptWitnessForSpending Era` for now, right? i.e. we could generalize "our helper" `mkScriptWitness` when we need to and then, this might already be merged upstream?

Reply:
Well, since we already know we are going to need script witnesses for minting, I went for the generalized version right away (which isn't much more work to be honest...).